main/fopen_wrappers.c

```diff
@@ -603,7 +603,10 @@ PHPAPI zend_string *php_resolve_path(const char *filename, size_t filename_lengt
 		const char *exec_fname = ZSTR_VAL(exec_filename);
 		size_t exec_fname_length = ZSTR_LEN(exec_filename);
 
-		while ((--exec_fname_length < SIZE_MAX) && !IS_SLASH(exec_fname[exec_fname_length]));
+		while (exec_fname_length != 0 && !IS_SLASH(exec_fname[exec_fname_length])) {
```

Comment:
Shouldn't this be:
```suggestion
		while (exec_fname_length != 0 && !IS_SLASH(exec_fname[exec_fname_length - 1])) {
```

or am I missing something?

Reply:
no you re right I forgot to move the offset logic entirely